src/Compilers/Core/Portable/SourceGeneration/Nodes/IncrementalValueSources.cs

```diff
@@ -75,11 +87,13 @@ public InputNode<ISyntaxContextReceiver> GetOrCreateReceiverNode()
                 return InterlockedOperations.Initialize(ref _receiverNode, new InputNode<ISyntaxContextReceiver>());
```

Comment:
>new InputNode<ISyntaxContextReceiver>() [](http://example.com/codeflow?start=75&length=39)

We're creating a `new InputNode<ISyntaxContextReceiver>()` each time `GetOrCreateReceiverNode()` is called.